retail/api/grs-search/SearchWithQueryExpansion.cs

```diff
@@ -0,0 +1,119 @@
+// Copyright 2021 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// [START retail_search_for_products_with_query_expansion_specification]
+// Call Retail API to search for a products in a catalog,
+// enabling the query expansion feature to let the Google Retail Search build an automatic query expansion.
+
+using Google.Cloud.Retail.V2;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Serialization;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace grs_search.search
+{
+    public static class SearchWithQueryExpansion
+    {
+        private const string Endpoint = "retail.googleapis.com";
```

Comment:
If this is the default endpoint you dont need to specify it.

Reply:
It was a recommended by the Lei Chen - the Retail libraries dev manager to directly specify the endpoint to avoid authorization issues